Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -364,6 +364,231 @@ private final class TokenStreamCreator: SyntaxVisitor {
     return false
   }
 
+  // MARK: - Control flow statement nodes
+
+  override func visit(_ node: IfStmtSyntax) {
+    before(node.ifKeyword, tokens: .open(.inconsistent, 3))
+    after(node.ifKeyword, tokens: .space)
+    before(node.body.leftBrace, tokens: .break(offset: -3), .close)
+
+    if !areBracesCompletelyEmpty(node.body, contentsKeyPath: \.statements) {
+      after(node.body.leftBrace, tokens: .break(offset: 2), .open(.consistent, 0))
+      before(node.body.rightBrace, tokens: .break(offset: -2), .close)
+    } else {
+      before(node.body.rightBrace, tokens: .break(size: 0))
+    }
+
+    before(node.elseKeyword, tokens: .break)
+    after(node.elseKeyword, tokens: .break)
+
+    if let elseBody = node.elseBody as? CodeBlockSyntax {
+      if !areBracesCompletelyEmpty(elseBody, contentsKeyPath: \.statements) {
```

Comment:
What does the backslash mean in this situation?

Reply:
It starts a "key path", which is a way of abstracting a property access.

Not all braced blocks expose the same properties for the collection of things inside the braces. For example, an `if` statement has a `statements` property which is a list of nodes, but a `switch` statement has a `cases` property. In order to check whether the collection is empty, I can pass the node in and the key path describing the property it should check, and the helper function resolves it "dynamically".

Some more info if you're curious: https://www.swiftbysundell.com/posts/the-power-of-key-paths-in-swift